app/frontend/wallet/shelley/shelley-bitbox02-crypto-provider.ts

```diff
@@ -0,0 +1,350 @@
+      return await f(_activeBitBox02)
+    }
+    try {
+      const devicePath = await bitbox02API.getDevicePath()
+      _activeBitBox02 = new bitbox02API.BitBox02API(devicePath)
+      await _activeBitBox02.connect(
+        (pairingCode) => {
+          config.bitbox02OnPairingCode(pairingCode)
+        },
+        () => {
+          config.bitbox02OnPairingCode(null)
+          return Promise.resolve()
+        },
+        (attestationResult) => {
+          debugLog(`BitBox02 attestation: ${attestationResult}`)
+        },
+        () => {
+          _activeBitBox02 = null
+        },
+        (status) => {
+          if (status === bitbox02Constants.Status.PairingFailed) {
+            config.bitbox02OnPairingCode(null)
+          }
+        }
+      )
+
+      if (_activeBitBox02.firmware().Product() !== bitbox02Constants.Product.BitBox02Multi) {
+        throw new Error('Unsupported device')
+      }
+
+      return await f(_activeBitBox02)
+    } catch (err) {
+      debugLog(err)
+      if (_activeBitBox02 !== null) {
+        try {
+          _activeBitBox02.close()
+        } finally {
+          _activeBitBox02 = null
+        }
+      }
+      throw new InternalError(InternalErrorReason.BitBox02Error, {
+        message: err,
+      })
+    }
+  }
+
+  const derivationScheme = derivationSchemes.v2
+
+  const bb02Network = {
+    [NetworkId.MAINNET]: bitbox02Constants.messages.CardanoNetwork.CardanoMainnet,
+    [NetworkId.TESTNET]: bitbox02Constants.messages.CardanoNetwork.CardanoTestnet,
+  }[network.networkId]
+
+  const version = await withDevice((bitbox02: BitBox02API) => {
+    const version = bitbox02.version().split('.')
+    return Promise.resolve({
+      major: version[0],
+      minor: version[1],
+      patch: version[2],
+    })
+  })
+
+  const getVersion = (): string => `${version.major}.${version.minor}.${version.patch}`
+
+  ensureFeatureIsSupported(CryptoProviderFeature.MINIMAL)
+
+  const getType = () => CryptoProviderType.BITBOX02
+
+  const deriveXpub = CachedDeriveXpubFactory(
+    derivationScheme,
+    config.shouldExportPubKeyBulk && isFeatureSupported(CryptoProviderFeature.BULK_EXPORT),
+    async (derivationPaths: BIP32Path[]) => {
+      return await withDevice(async (bitbox02: BitBox02API) => {
+        const xpubs: Uint8Array[] = await bitbox02.cardanoXPubs(derivationPaths)
+        return xpubs.map(Buffer.from)
+      })
+    },
+    isFeatureSupported(CryptoProviderFeature.BYRON)
+  )
+
+  function isFeatureSupported(feature: CryptoProviderFeature): boolean {
+    switch (feature) {
+      case CryptoProviderFeature.BYRON:
+      case CryptoProviderFeature.POOL_OWNER:
+      case CryptoProviderFeature.MULTI_ASSET:
+      case CryptoProviderFeature.VOTING:
+        return false
+      default:
+        return BITBOX02_VERSIONS[feature]
+          ? hasRequiredVersion(version, BITBOX02_VERSIONS[feature])
+          : hasRequiredVersion(version, BITBOX02_VERSIONS[CryptoProviderFeature.MINIMAL])
+    }
```

Comment:
```suggestion
        return hasRequiredVersion(version, BITBOX02_VERSIONS[feature] ?? BITBOX02_VERSIONS[CryptoProviderFeature.MINIMAL])
    }
```

same as a few lines below